adding do_not_fail_on_forbidden section to docs #4896 (#6958)
opensearch-project/documentation-website: _security/access-control/permissions.md

Keep in mind that enabling this feature and mapping system index permissions to normal users gives those users access to indexes that may contain sensitive information and configurations essential to a cluster's health. We also recommend caution when mapping users to `restapi:admin/roles` because this permission gives a user not only the ability to assign the system index permission to another user but also the ability to self-assign access to any system index.
{: .warning }

### `do_not_fail_on_forbidden`

> If the user attempts to query multiple indexes, some of which they lack permission for, by default they get an `error` in OpenSearch Dashboards UI or an `exception` using `cURL` or API. If instead, you want the user to receive the results from the search for any of the indexes they _do_ have permission for, you can set the option `do_not_fail_on_forbidden` to `true` in `config.yml`. See the following example:

Reviewer: ```suggestion
If a user attempts to query multiple indexes, some of which they lack permissions for, by default they get an `error` in the OpenSearch Dashboards UI or an `exception` when using `cURL` or an API. If you instead want the user to receive the search results for any of the indexes for which they _do_ have permissions, you can set the option `do_not_fail_on_forbidden` to `true` in `config.yml`. See the following example:
```

> If the user attempts to query multiple indices, some of which they lack permission for, by default they will get an `error` in OpenSearch Dashboards UI, or an `exception` using `cURL` or API. If instead, you want the user to receive the results from the search for any of the indices they _do_ have permission for, you can set the option `do_not_fail_on_forbidden` to `true` in `config.yml`. See the following example:

Reviewer 2: ```suggestion
If the user attempts to query multiple indexes, some of which they lack permission for, by default they will get an `error` in OpenSearch Dashboards UI, or an `exception` using `cURL` or API. If instead you want the user to receive the results from the search for any of the indices they _do_ have permission for, you can set the option `do_not_fail_on_forbidden` to `true` in `config.yml`. See the following example:
```

> If the user attempts to query multiple indexes, some of which they do not have permission for, by default they will get an `error` in OpenSearch Dashboards UI or `exception` using `cURL` or API (failing to return the result for indexes which they **do** have access to). If you want the users to receive the hits from the search for only the indexes they have permission for, you can set the option `do_not_fail_on_forbidden` to `true` in `config.yml`. See the following example:

Reviewer 3: ```suggestion
If the user attempts to query multiple indices, some of which they lack permission for, by default they will get an `error` in OpenSearch Dashboards UI, or an `exception` using `cURL` or API. If instead, you want the user to receive the results from the search for any of the indices they _do_ have permission for, you can set the option `do_not_fail_on_forbidden` to `true` in `config.yml`. See the following example:
```

Reviewer 4: @scrawfor99 - We actually DO use indexes, not indices, unless we are referring to the code itself. :)

```
_meta:
  type: "config"
  config_version: 2
config:
  dynamic:
    http:
      anonymous_auth_enabled: false
      xff:
        enabled: false
        internalProxies: "192\\.168\\.0\\.10|192\\.168\\.0\\.11"
    do_not_fail_on_forbidden: true
    authc:
      basic_internal_auth_domain:
      ...
```

> It is important to remember that if this option is set to `true`, the user is served the data as if it is the complete set. There is no indication that some data may be omitted.

Reviewer: ```suggestion
It is important to remember that if this option is set to `true`, then the user is served the data as if it is the complete set. There is no indication that some data may be omitted.
```

> It is important to remember that if this option is set to `true`, the user is served the data as if it is the complete set. There is no indication that some data may be omitted.

Reviewer: "dataset" instead of "set"?

> It is important to remember that if this option is set to `true`, then the user is served the data as if it is the complete set. There is no indication that some data may be omitted.

Reviewer 2: ```suggestion
It is important to remember that if this option is set to `true`, then the user is served the data as if it is the complete dataset. There is no indication that some data may be omitted.
```

> It is important to remember that if this option is set to `true`, the user will assume that the data returned is the complete set, this is also true for any aggregation.

Reviewer 3: ```suggestion
It is important to remember that if this option is set to `true`, the user will be served the data as if it is the complete set. There is no indication that some data may be omitted.
```

{: .warning }

### `do_not_fail_on_forbidden_empty`

> When the user views a visualization for which they lack index permissions, they see `error` in place of the visualization. To change this behavior to display `No results displayed because all values equal 0.`, you can set `do_not_fail_on_forbidden_empty` to `true` in `config.yml`. This option is only valid if `do_not_fail_on_forbidden` is also set to `true`. See the following example:

Reviewer: ```suggestion
When a user attempts to view a visualization for which they lack index permissions, they will see `error` in place of the visualization. To change this behavior to display `No results displayed because all values equal 0.`, you can set `do_not_fail_on_forbidden_empty` to `true` in `config.yml`. This option is only valid if `do_not_fail_on_forbidden` is also set to `true`. See the following example:
```

> When the user views a visualization which they lack index permissions for, they will see `error` in place of the visualization. To change this behavior to display `No results displayed because all values equal 0.` you can set `do_not_fail_on_forbidden_empty` to `true` in `config.yml`. This option is only valid if `do_not_fail_on_forbidden` is also set to `true`. See the following example:

Reviewer 2: ```suggestion
When the user views a visualization for which they lack index permissions, they will see `error` in place of the visualization. To change this behavior to display `No results displayed because all values equal 0.`, you can set `do_not_fail_on_forbidden_empty` to `true` in `config.yml`. This option is only valid if `do_not_fail_on_forbidden` is also set to `true`. See the following example:
```

> When the user views visualization which returns empty result due to lack of permissions, they will see `error` in place of the visualization. To change this behavior to display `No results displayed because all values equal 0.` you can set `do_not_fail_on_forbidden_empty` to `true` in `config.yml`. This option is only valid if `do_not_fail_on_forbidden` is also set to `true`. See the following example:

Reviewer 3: ```suggestion
When the user views a visualization which they lack index permissions for, they will see `error` in place of the visualization. To change this behavior to display `No results displayed because all values equal 0.` you can set `do_not_fail_on_forbidden_empty` to `true` in `config.yml`. This option is only valid if `do_not_fail_on_forbidden` is also set to `true`. See the following example:
```

| `indices:data/write/bulk*` |  Permission to run a bulk request. |
| `indices:data/write/delete` |  Permission to [delete documents]({{site.url}}{{site.baseurl}}/api-reference/document-apis/delete-document/). |
| `indices:data/write/delete/byquery` |  Permission to delete all documents that [match a query]({{site.url}}{{site.baseurl}}/api-reference/document-apis/delete-by-query/). |

> | `indices:data/write/plugins/replication/changes` |  Permission to make changes to data replication configurations and settings within indexes. |

Reviewer: ```suggestion
| `indices:data/write/plugins/replication/changes` |  Permission to change data replication configurations and settings within indexes. |
```

## Security REST permissions

Allowing access to these endpoints has the potential to trigger operational changes in the cluster. Proceed with caution.
{: .warning }

> The following permissions apply to REST APIs to control access to the endpoints. Granting access to any of these APIs allows a user the permission to change fundamental operational components of the Security plugin:

Reviewer: ```suggestion
The following REST API permissions control access to the endpoints. Granting access to any of these APIs allows a user to change fundamental operational components of the Security plugin:
```
